src/replication/log_consumer.cpp

```diff
@@ -164,49 +132,42 @@ namespace cubreplication
     public:
       dispatch_daemon_task (log_consumer &lc)
 	: m_filtered_apply_end (log_Gl.hdr.m_ack_stream_position)
+	, m_entry_fetcher (*lc.get_stream ())
 	, m_lc (lc)
+	, m_stop (false)
       {
-      }
-
-      bool is_filtered_apply_segment (cubstream::stream_position stream_entry_end) const
-      {
-	return stream_entry_end <= m_filtered_apply_end;
-      }
-
-      bool filter_out_stream_entry (stream_entry *repl_stream_entry) const
-      {
-	if (is_filtered_apply_segment (repl_stream_entry->get_stream_entry_end_position ()))
-	  {
-	    MVCCID mvccid = repl_stream_entry->get_mvccid ();
-	    return repl_stream_entry->is_group_commit ()
-		   || log_Gl.m_repl_rv.m_active_mvcc_ids.find (mvccid) == log_Gl.m_repl_rv.m_active_mvcc_ids.end ();
-	  }
-
-	if (!log_Gl.m_repl_rv.m_active_mvcc_ids.empty ())
-	  {
-	    log_Gl.m_repl_rv.m_active_mvcc_ids.clear ();
-	  }
-	return false;
+	auto on_fetch_func = [this] (stream_entry *se, bool & should_skip)
+	{
+	  if (se->is_group_commit ())
+	    {
+	      se->unpack ();
+	      assert (se->get_stream_entry_end_position () > se->get_stream_entry_start_position ());
+	      m_lc.ack_produce (se->get_stream_entry_end_position ());
+	    };
+	  should_skip = filter_out_stream_entry (se);
```

Comment:
`should_skip = filter_out_stream_entry (se);` check can be moved to execute function.

EDIT: actually the entire code of on_fetch_func can be moved to execute function. If we don't consider how consumer previously worked, is there a reason to separate these operations from other operations in execute function? It these are not inherently part of fetching process, it is better to keep them with other operations from dispatch execution.